### PR TITLE
enhance ps

### PR DESCRIPTION
**Pull request by marxin**
New output:

```
PID     PPID    CPU  ST COMM
1       0       4    S  systemd
2       0       13   S  [kthreadd]
3       2       0    I  [rcu_gp]
4       2       0    I  [rcu_par_gp]
5       2       0    I  [slub_flushwq]
6       2       0    I  [netns]
8       2       0    I  [kworker/0:0H]
10      2       0    I  [mm_percpu_wq]
11      2       0    I  [rcu_tasks_kthre]
...
```

Signed-off-by: Martin Liska <mliska@suse.cz>

### CODE REVIEW

**Comment by marxin** [comment]
I'm playing with the `drgn` and I might enhance some scripts in `contrib`. Are pull requests welcomed in the area?

File: contrib/ps.py [hunk 1]
```diff
@@ -5,9 +5,28 @@
...
+from drgn.helpers.linux.sched import task_cpu, task_state_to_char
 
-print("PID        COMM")
+
+def is_kthread(task):
+    """
+    Make a guess if task_struct is a kernel thread.
+    """
+
+    return task.mm.value_() == 0
```
Comment (osandov) [inline contrib/ps.py:16]: Tiny suggestion:

```suggestion
    return not task.mm
```

**Review by osandov — COMMENTED** [review]
Yup, I'm happy to accept PRs in contrib! In case you haven't seen it, [contrib/README.rst](https://github.com/osandov/drgn/blob/main/contrib/README.rst) has some background.

**Comment by marxin** [comment]
> Yup, I'm happy to accept PRs in contrib! In case you haven't seen it, [contrib/README.rst](https://github.com/osandov/drgn/blob/main/contrib/README.rst) has some background.

Sounds good!

**Pull request merged**